test/FullRange.t.sol

```diff
@@ -0,0 +1,767 @@
+        (, address liquidityToken) = fullRange.poolInfo(withLiqId);
+
+        IPoolManager.SwapParams memory params =
+            IPoolManager.SwapParams({zeroForOne: true, amountSpecified: 1 ether, sqrtPriceLimitX96: SQRT_RATIO_1_2});
+
+        PoolSwapTest.TestSettings memory testSettings =
+            PoolSwapTest.TestSettings({withdrawTokens: true, settleUsingTransfer: true});
+
+        swapRouter.swap(withLiqKey, params, testSettings);
+
+        UniswapV4ERC20(liquidityToken).approve(address(fullRange), type(uint256).max);
+
+        address token0Addr = address(token0);
+        address token2Addr = address(token2);
+
+        snapStart("FullRangeRemoveLiquidityAndRebalance");
+        fullRange.removeLiquidity(FullRange.RemoveLiquidityParams(token0Addr, token2Addr, 3000, 5 ether, MAX_DEADLINE));
+        snapEnd();
+
+        (bool hasAccruedFees,) = fullRange.poolInfo(withLiqId);
+        assertEq(hasAccruedFees, false);
+    }
+
+    function testFullRange_removeLiquidity_RemoveAllFuzz(uint256 amount) public {
+        manager.initialize(key, SQRT_RATIO_1_1);
+        (, address liquidityToken) = fullRange.poolInfo(id);
+
+        if (amount < LOCKED_LIQUIDITY) {
+            vm.expectRevert(FullRange.LiquidityDoesntMeetMinimum.selector);
+            fullRange.addLiquidity(
+                FullRange.AddLiquidityParams(
+                    address(token0), address(token1), 3000, amount, amount, amount, amount, address(this), MAX_DEADLINE
+                )
+            );
+        } else if (amount >= MAX_TICK_LIQUIDITY) {
+            vm.expectRevert();
+            fullRange.addLiquidity(
+                FullRange.AddLiquidityParams(
+                    address(token0), address(token1), 3000, amount, amount, amount, amount, address(this), MAX_DEADLINE
+                )
+            );
+        } else {
+            fullRange.addLiquidity(
+                FullRange.AddLiquidityParams(
+                    address(token0), address(token1), 3000, amount, amount, 0, 0, address(this), MAX_DEADLINE
+                )
+            );
+
+            // Test contract removes liquidity, succeeds
+            UniswapV4ERC20(liquidityToken).approve(address(fullRange), type(uint256).max);
+
+            if (amount > UniswapV4ERC20(liquidityToken).balanceOf(address(this))) {
+                vm.expectRevert();
+                fullRange.removeLiquidity(
+                    FullRange.RemoveLiquidityParams(address(token0), address(token1), 3000, amount, MAX_DEADLINE)
+                );
+            } else {
+                fullRange.removeLiquidity(
+                    FullRange.RemoveLiquidityParams(address(token0), address(token1), 3000, amount, MAX_DEADLINE)
+                );
+            }
+
+            assertTrue(manager.getLiquidity(id) >= LOCKED_LIQUIDITY);
+        }
+    }
+
+    function testFullRange_removeLiquidity_ThreeLPsRemovePrincipalAndFees() public {
```

Comment:
so for this test, can we test that all LP's remove their liquidity, and that the only liquidity left in the pool is the MINIMUM liquidity?? I think that's the big important thing, bc to me it still looks like we're not removing the extra fees in the hook function?